dalle2_pytorch/train_diffusion_prior.py

```diff
@@ -0,0 +1,160 @@
+import argparse
+import os
+from dalle2_pytorch import DiffusionPrior
+from embedding_reader import EmbeddingReader
+from dalle2_pytorch import DiffusionPriorNetwork
+import numpy as np
+import math
+from tqdm import tqdm
+import torch
+from torch import nn
+import wandb
+os.environ["WANDB_SILENT"] = "true"
+
+def train(image_embed_dim,
+          image_embed_url,
+          text_embed_url,
+          batch_size,
+          train_percent,
+          val_percent,test_percent,
+          num_epochs,
+          loss_type,
+          condition_on_text_encodings,
+          device,
+          learning_rate=0.01):
+    # DiffusionPriorNetwork 
+    prior_network = DiffusionPriorNetwork( dim = image_embed_dim, depth = 6, dim_head = 64, heads = 8).to(device)
+    
+    # DiffusionPrior with text embeddings and image embeddings pre-computed
+    diffusion_prior = DiffusionPrior( net = prior_network, clip = None, image_embed_dim = image_embed_dim, 
+                                     timesteps = 100, cond_drop_prob = 0.2, loss_type = loss_type,
+                                     condition_on_text_encodings = condition_on_text_encodings).to(device)
+    # Gtext_reader image and text embeddings from the servers
+    print("==============Downloading embeddings - image and text====================")
+    image_reader = EmbeddingReader(embeddings_folder=image_embed_url, file_format="npy")
+    text_reader = EmbeddingReader(embeddings_folder=text_embed_url, file_format="npy")
+
+    ### Training code ###
+    optimizer = torch.optim.SGD(diffusion_prior.parameters(), lr = learning_rate)
+    epochs = num_epochs
+    min_valid_loss = np.inf
+    for e in range(epochs):
+        train_loss = 0.0 
+        print("Training loop - epoch number ",e)
+        train_set_size = int(train_percent*image_reader.count)
+        for emb_images,emb_text in zip(image_reader(batch_size=batch_size, start=0, end=train_set_size),
+                text_reader(batch_size=batch_size, start=0, end=train_set_size)):
+            emb_images = list(emb_images)
+            emb_text = list(emb_text)
+            print(emb_images[0].shape,emb_text[0].shape)
+            if torch.cuda.is_available():
```

Comment:
since you already set the device based on this logic above, do you need this again?

Reply:
Fixed.